src/main/java/com/google/api/generator/engine/ast/MethodDefinition.java

```diff
@@ -277,15 +277,14 @@ public MethodDefinition build() {
         if (method.returnExpr() != null && !isLastStatementThrowExpr) {
           if (method.returnType().isPrimitiveType()) {
             Preconditions.checkState(
-                method.returnExpr().type().isPrimitiveType()
-                    && method.returnType().equals((method.returnExpr().type())),
-                "Method primitive return type does not match the return expression type");
+                method.returnType().equals((method.returnExpr().type())),
+                "Method primitive return type is not equal to the return expression type");
```

Comment:
Would prefer keeping the original message, due to the actual semantics of `equals()`.

Reply:
Thanks, the advice below sounds reasonable to me, changed it back.